documentation/design/proposals/iot-console-backend.adoc

```diff
@@ -0,0 +1,64 @@
+== Overview
+
+The iot console backend should provide a graphQL api that allows the console UI to :
+* list, create, update, delete iot projects.
+* list, create, update, delete devices and their credentials in iot projects the user have access to.
+* list addresses and connection details of the downstream messaging endpoints.
+* list hono protocols adapter the user have access to, and their connection details.
+* get metrics for iot projects and devices.
+
+
+== Design
+
+The above requirements involves two kind of resources :
+* kubernetes resources.
+* Hono resources that are accessed through the device registry service.
+
+=== Kubernetes Object
+
+The existing console back end will be updated to access the relevant kubernetes objects
+(iot projects, routes for protocol adapters, messaging endpoint).
+The existing approach of permissions will be used : it is the backend's responsibility to
+filter the user's view so it fits the user's permissions.
+For write operations, console-server propagates the user's bearer token made available by oauth-proxy.
+
+=== Hono devices & Credentials
+
+The iot devices and their credentials needs to be accessed through the device registry REST api.
+Therefore, a rest client must be used in the console backend. For this part, the internal cache
+of the console-backend will not be used. Also, the token verification can be ignored, as the device registry
+already verify the bearer token in HTTP requests.
+In this scope the backend acts as a simple REST to graphQL translator.
+
+
+=== Metrics
+
+Metrics will also be scraped directly from the agents.\
+Specifics metrics needs are TBD.
+
+==== GraphQL queries
+
+Get all user-visible iotprojects :
+`iotProjects()`
+
+Get all adapters accessible with the given iotproject
+`adapters(iotproject: String!)`
+
+Get all user-visible devices in the given iot project, paginated
```

Comment:
Not sure if this has already been covered, we would require a 'total' (total number of items in the list) in case of all paginated lists.

Reply:
I added a total in a dedicated query result type

Reply:
Great, thanks a lot!